Assets/BossRoom/Scripts/Client/Game/State/ClientCharSelectState.cs

```diff
@@ -373,9 +385,50 @@ private void ConfigureUIForLobbyMode(LobbyMode mode)
         /// <summary>
         /// Called directly by UI elements!
         /// </summary>
+        /// <remarks>
+        /// For a reactive character select screen, the local player's selection is presented instantaneously
+        /// before a NetworkListEvent can be received from the server.
+        /// To accomplish this, the previous seat selected will be cleared or re-populated by a shared player, and
+        /// the new seat will be populated with the local selection.
+        /// </remarks>
         /// <param name="seatIdx"></param>
         public void OnPlayerClickedSeat(int seatIdx)
         {
+            if (m_LastSeatSelected != -1)
+            {
+                var otherPlayerSharesSeat = false;
+
+                // if any other user shares seat, set them as owner; else deactivate seat
```

Comment:
"owner"? I thought the lockedIn state was server driven?

Reply:
When a player chooses a new seat, I need to clear the local player's previous seat selection from the UI. But sometimes that seat is shared by another player.

Whether that previous seat now becomes empty, or "held" by another player is determined by the last update that was received by the server on the seats.